--- a/core/Controller/LoginController.php
+++ b/core/Controller/LoginController.php
@@ -119,7 +119,10 @@
 		}
 		$this->userSession->logout();

-		$response = new RedirectResponse($this->urlGenerator->linkToRouteAbsolute('core.login.showLoginForm'));
+		$redirectUrl = $this->config->getSystemValue('logout_url',
+			$this->urlGenerator->linkToRouteAbsolute('core.login.showLoginForm')
+		);
+		$response = new RedirectResponse($redirectUrl);
 		$response->addHeader('Clear-Site-Data', '"cache", "cookies", "storage", "executionContexts"');
 		return $response;
 	}
