diff --git a/core/Controller/LoginController.php b/core/Controller/LoginController.php
index 85d3b6b837..e7b9f5fdca 100644
--- a/core/Controller/LoginController.php
+++ b/core/Controller/LoginController.php
@@ -130,11 +130,12 @@ class LoginController extends Controller {
 		}
 		$this->userSession->logout();
 
-		$response = new RedirectResponse($this->urlGenerator->linkToRouteAbsolute(
-			'core.login.showLoginForm',
-			['clear' => true] // this param the the code in login.js may be removed when the "Clear-Site-Data" is working in the browsers
+		$redirectUrl = $this->config->getSystemValue('logout_url',
+			$this->urlGenerator->linkToRouteAbsolute(
+				'core.login.showLoginForm',
+				['clear' => true] // this param the the code in login.js may be removed when the "Clear-Site-Data" is working in the browsers)
 		));
-
+		$response = new RedirectResponse($redirectUrl);
 		$this->session->set('clearingExecutionContexts', '1');
 		$this->session->close();
 		$response->addHeader('Clear-Site-Data', '"cache", "storage", "executionContexts"');
@@ -329,7 +330,7 @@ class LoginController extends Controller {
 				['app' => 'core']);
 			return $this->createLoginFailedResponse($user, $originalUser,
 				$redirect_url, self::LOGIN_MSG_INVALIDPASSWORD);
-		}
+			}
 
 		// TODO: remove password checks from above and let the user session handle failures
 		// requires https://github.com/owncloud/core/pull/24616
